### PR TITLE
feat(Stepper): horizontal

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
Horizontal variation was missing

**What is the chosen solution to this problem?**
Add it

**Please check if the PR fulfills these requirements**

- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: src/components/Stepper/Step/variations/Step.progress.tsx [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
-import styled from 'styled-components';
+import styled, { keyframes, ThemeProps } from 'styled-components';
 import Step from '../Step';
 import tokens from '../../../../tokens';
 
+const pulse = ({ theme }: ThemeProps<any>) => keyframes`
+	0% {
+		transform: scale(0.95);
+		box-shadow: 0 0 0 0 ${theme.colors?.activeColor[100]};
+}
```
Comment (ncomont) [inline src/components/Stepper/Step/variations/Step.progress.tsx:10]: (indentation is weird)

File: src/components/Stepper/Stepper.tsx [hunk 2]
```diff
@@ -1,20 +1,39 @@
 import React, { PropsWithChildren } from 'react';
+
 import * as S from './Stepper.style';
 
-export type StepperProps = PropsWithChildren<any>;
+export type StepperProps = PropsWithChildren<any> & {
+	orientation: 'horizontal' | 'vertical';
```
Comment (ncomont) [inline src/components/Stepper/Stepper.tsx:6]: maybe we could use and export consts ?

File: src/components/Stepper/Progress/Progress.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import VisuallyHidden from '../../VisuallyHidden';
+
+import * as S from './Progress.style';
+
+export type ProgressProps = React.PropsWithChildren<any> & {
+	orientation: 'horizontal' | 'vertical';
+	valuemin?: number;
```
Comment (ncomont) [inline src/components/Stepper/Progress/Progress.tsx:8]: why didn't you used camel case, or simply `min`, `max`, `value` to fit the HTML progress api ?

File: src/components/Stepper/Stepper.tsx [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
-						{React.cloneElement(child, { 'data-index': index + 1 })}
-						<div aria-hidden className="stepper__progress-bar" />
-					</S.StepperStep>
-				))}
-		</S.Stepper>
-	),
+	({ children, orientation, loading, ...rest }: StepperProps, ref) => {
+		const isInProgress = (child: React.ReactNode) =>
+			// @ts-ignore
+			child.type.type.displayName.includes('InProgress');
```
Comment (ncomont) [inline src/components/Stepper/Stepper.tsx:14]: can you use typeof or something like that ?

File: src/components/Stepper/Progress/Progress.tsx (hunk 3 above)
Comment (frassinier) [reply src/components/Stepper/Progress/Progress.tsx:8]: I was using aria default attributes, but it makes sense to use default `<Progress>` attrs 😅

**Review by ncomont — APPROVED** [review]

**Pull request merged**